Source/Scene/WebMapTileServiceImageryProvider.js

```diff
@@ -372,6 +380,8 @@ function requestImage(imageryProvider, col, row, level, request, interval) {
       queryParameters: query,
       request: request,
     });
+    //When useKvp is true, replace the {s} keyword in the requested URL. QUESTIONS: modify `resource`? Should I replace {s} with ""?
```

Comment:
For questions like this, it's usually better to include them as self-review comments on your PR. This makes it easier to track conversations about questions that come up.

As for your question, I don't think this is right, since we want to use the subdomains in the url if they're provided. See `"supports subdomains string urls"` in the specs for this file - currently it passes on your branch because the url is just an empty string, so `indexOf` returns 0. But, on master, the url is `"1"`, which matches the subdomain template... I wonder if the specs should be more rigorous. 

I'm not that familiar with this code, but I think the fix here should do something like this if `subdomains.length > 0`: 
```js
var templateValues = {
  s: subdomains[(col + row + level) % subdomains.length],
};

resource = imageryProvider._resource.getDerivedResource({
  queryParameters: query,
  request: request,
});
resource.setTemplateValues(templateValues);
```

@tfili Could you please confirm if this is the right idea? In Sandcastle the above change appears to make the expected request urls but I wanted to double check.

Reply:
Thank you for your feedback on my in-code comments. While I found them helpful, I can see how they make the codebase more messy. I removed the majority of them.

I support making the specs more rigorous, as it seems plausible that people in the future might have a similar point of confusion. 

@ebogo1 I am not completely sure where your code snippet should be added. I am assuming that this code snippet is used when `imageryProvider._useKvp` is `true`? Regardless, I am curious to hear what @tfili thinks.

Reply:
It looks like Fili had the same idea in his comment about the spec so I continued the discussion there.